src/Tree/HydroTree.cpp

```diff
@@ -1449,17 +1449,63 @@ void HydroTree<ndim,ParticleType>::FindMpiTransferParticles
     Box<ndim>& nodebox = mpinodes[inode].domain;
 
     // Start from root-cell
-    int NumPartFound = tree->FindBoxOverlapParticles(nodebox, all_particles_to_export,
+    tree->FindBoxOverlapParticles(nodebox, particles_to_export[inode],
                                                      hydro->GetParticleArray()) ;
 
     // Copy particles to per processor array
-    if (NumPartFound > 0)
-      particles_to_export[inode].insert(particles_to_export[inode].end(),
-                                        all_particles_to_export.end() - NumPartFound,
-                                        all_particles_to_export.end());
+    all_particles_to_export.insert(all_particles_to_export.end(),
+                                        particles_to_export[inode].begin(),
+                                        particles_to_export[inode].end());
   }
   //-----------------------------------------------------------------------------------------------
 
+#ifndef NDEBUG
+  VerifyUniqueIds(all_particles_to_export.size(),hydro->Nhydro,&all_particles_to_export[0]);
```

Comment:
Oh, we definitely do NOT want VERIFY_ALL when using Travis since that will kill us.  Yes, this particular validation is a special brute-force check that we don't want in general.  Maybe it's best to put it in the 'higher' debug flags (i.e. 2).  

Surely we only want production code on Travis?  We only need the debug options (with asserts et al.) when there's actually a problem we want to get to the bottom of!